tool/tctl/common/decision/command.go

```diff
@@ -0,0 +1,78 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package decision
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	decisionpb "github.com/gravitational/teleport/api/gen/proto/go/teleport/decision/v1alpha1"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	commonclient "github.com/gravitational/teleport/tool/tctl/common/client"
+	tctlcfg "github.com/gravitational/teleport/tool/tctl/common/config"
+)
+
+// Command is a group of commands to interact with the Teleport Decision Service.
+type Command struct {
+	// Output is the writer that any command output should be written to.
+	Output io.Writer
+
+	evaluateSSHCommand      EvaluateSSHCommand
+	evaluateDatabaseCommand EvaluateDatabaseCommand
+}
+
+// Initialize sets up the "tctl decision" command.
+func (c *Command) Initialize(app *kingpin.Application, _ *tctlcfg.GlobalCLIFlags, _ *servicecfg.Config) {
+	if c.Output == nil {
+		c.Output = os.Stdout
+	}
+
+	cmd := app.Command("decision", "Interact with the Teleport Decision Service.").Hidden()
+	c.evaluateSSHCommand.Initialize(cmd, c.Output)
+	c.evaluateDatabaseCommand.Initialize(cmd, c.Output)
+}
+
+// Client contains methods required by this command
+// to interact with the control plane.
+type Client interface {
+	DecisionClient() decisionpb.DecisionServiceClient
+}
+
+// TryRun attempts to run subcommands.
+func (c *Command) TryRun(ctx context.Context, cmd string, clientFunc commonclient.InitFunc) (bool, error) {
+	var run func(context.Context, Client) error
```

Comment:
Why not avoid the interface altogether and pass in `client.DecisionClient()` instead?

```suggestion
	var run func(context.Context, decisionpb.DecisionServiceClient) error
```